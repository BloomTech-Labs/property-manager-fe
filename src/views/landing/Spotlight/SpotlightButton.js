import React from 'react';
import { navigate } from '@reach/router';

const SpotlightButton = () => {
  return (
    <div className="btnWrapper">
      <button
        className="spotBtn"
        type="submit"
        onClick={() => navigate('/signup')}
      >
        Sign up
      </button>
    </div>
  );
};

export default SpotlightButton;
