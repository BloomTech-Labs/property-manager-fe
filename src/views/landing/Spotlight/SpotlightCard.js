import DocuImage from '../../../assets/svg/landing-docu-card.svg';
import PayImage from '../../../assets/svg/landing-stripe.svg';
import WorkImage from '../../../assets/svg/landing-workorder.svg';
import ChartImage from '../../../assets/svg/landing-chart.svg';

export default [
  {
    src: DocuImage,
    alt: 'Spotlight card with documents.',
    key: 'DOCUMENT_CARD',
    description:
      'Communicate and share documents with you tenants directly in the app.'
  },
  {
    src: PayImage,
    alt: 'Spotlight card highlghting Stripe.',
    key: 'PAY_CARD',
    description: 'Accept payments with confidence through Stripeâ€™s secure API.'
  },
  {
    src: WorkImage,
    alt: 'Spotlight card highlighting work orders.',
    key: 'WORK_CARD',
    description:
      'Organize and review repair requests from creation to completion.'
  },
  {
    src: ChartImage,
    alt: 'Spotlight card highlighting statistics and charts.',
    key: 'CHART_CARD',
    description:
      'Keep track of your cash flow month-to-month so you know exactly where your finances are.'
  }
];
