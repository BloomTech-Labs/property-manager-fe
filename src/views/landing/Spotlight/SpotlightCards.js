import React from 'react';
import images from './SpotlightCard';

const SpotlightCard = () => {
  return (
    <div className="spotlightCards">
      {images.map(({ src, alt, key, description }) => (
        <div key={key} className="spotlightCard">
          <img className="spotCardImages" src={src} alt={alt} />
          <hr className="spotDivider" />
          <h5 className="spotCardContent">{description}</h5>
        </div>
      ))}
    </div>
  );
};

export default SpotlightCard;
