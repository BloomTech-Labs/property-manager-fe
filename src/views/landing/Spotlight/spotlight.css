.spotlightWrapper {
  height: 100%;
  width: 100%;
  margin: 20vh auto;
}

@media (max-width: 1200px) {
  .spotlightWrapper {
    padding: 5%;
  }
}

@media (max-width: 1000px) {
  .spotlightWrapper {
    margin: auto;
    width: 90%;
  }
}

.spotHeadingWrapper {
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  width: 100%;
  margin: 2% auto;
}

@media (max-width: 768px) {
  .spotHeadingWrapper {
    flex-flow: column nowrap;
    width: 100%;
  }
}

.spotHeadingWrapper .spotHeaderImage {
  max-width: 500px;
  width: 100%;
  margin-right: 50px;
  z-index: 2;
}

@media (max-width: 1000px) {
  .spotHeadingWrapper .spotHeaderImage {
    width: 60%;
  }
}

@media (max-width: 768px) {
  .spotHeadingWrapper .spotHeaderImage {
    margin: 0 auto;
  }
}

@media (max-width: 500px) {
  .spotHeadingWrapper .spotHeaderImage {
    width: 60%;
  }
}

.spotHeadingWrapper .spotHeading {
  width: 100%;
  color: #2e3c50;
  font-family: Lato;
  font-size: 3rem;
  line-height: 53px;
}

@media (max-width: 1000px) {
  .spotHeadingWrapper .spotHeading {
    font-size: 2rem;
    line-height: 38px;
    text-align: center;
  }
}

@media (max-width: 768px) {
  .spotHeadingWrapper .spotHeading {
    font-size: 2rem;
    line-height: 38px;
    width: 70%;
    margin: 10px 0;
  }
}

@media (max-width: 500px) {
  .spotHeadingWrapper .spotHeading {
    width: 100%;
    margin-top: 15px;
    text-align: center;
    font-size: 1.6rem;
    line-height: 27px;
  }
}

.spotlightCards {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
}

@media (max-width: 1000px) {
  .spotlightCards {
    flex-flow: row wrap;
    justify-content: space-evenly;
    width: 70%;
    margin: 0 auto;
  }
}

@media (max-width: 768px) {
  .spotlightCards {
    width: 80%;
    margin: 0 auto;
  }
}

@media (max-width: 650px) {
  .spotlightCards {
    width: 100%;
  }
}

@media (max-width: 500px) {
  .spotlightCards {
    margin: 0 auto;
    flex-flow: column nowrap;
  }
}

.spotlightCards .spotlightCard {
  display: flex;
  flex-direction: column;
  background: #ffffff;
  align-items: center;
  padding: 1%;
  max-width: 225px;
  width: 100%;
  min-height: 300px;
  height: 100%;
  border: none;
  border-radius: 10px;
  box-shadow: -10px 4px 28px rgba(0, 0, 0, 0.25);
  margin: 20px 0;
  transition: all 0.3s ease;
}

.spotlightCards .spotlightCard:hover {
  transform: scale(1.05) rotate(-1deg);
}

.spotlightCards .spotlightCard .spotCardImages {
  padding: 5px;
  max-width: 175px;
  height: 150px;
  margin: auto;
  width: 100%;
}

.spotlightCards .spotlightCard .spotDivider {
  margin: 10px auto 0;
  width: 30%;
  border-top: 2px solid #6c63ff;
}

.spotlightCards .spotlightCard .spotCardContent {
  display: flex;
  align-items: center;
  margin: auto;
  width: 100%;
  font-size: 1rem;
  height: 100px;
  font-weight: 400;
  padding: 10px;
}

.btnWrapper {
  display: flex;
  justify-content: flex-end;
  margin: 70px 0 70px 70px;
}

@media (max-width: 1000px) {
  .btnWrapper {
    justify-content: center;
    margin: 70px auto;
  }
}

.btnWrapper .spotBtn {
  width: 290px;
  height: 50px;
  font-family: Lato;
  font-size: 1.2rem;
  font-weight: 500;
  box-shadow: -10px 4px 28px rgba(0, 0, 0, 0.25);
  border: 1px solid #6c63ff;
  border-radius: 6px;
  background: #ffffff;
  color: #6c63ff;
  text-align: center;
  cursor: pointer;
  transition: all 0.3s ease;
}

.btnWrapper .spotBtn:hover {
  background: #6c63ff;
  color: #fff;
}
