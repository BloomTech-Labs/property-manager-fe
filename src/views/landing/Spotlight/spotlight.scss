.spotlightWrapper {
  height: 100%;
  width: 100%;
  margin: 20vh auto;

  @media (max-width: 1200px) {
    padding: 5%;
  }

  @media (max-width: 1000px) {
    margin: auto;
    width: 90%;
  }
}

.spotHeadingWrapper {
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  width: 100%;
  margin: 2% auto;

  @media (max-width: 768px) {
    flex-flow: column nowrap;
    width: 100%;
  }

  .spotHeaderImage {
    max-width: 500px;
    width: 100%;
    margin-right: 50px;
    z-index: 2;

    @media (max-width: 1000px) {
      width: 60%
    }

    @media (max-width: 768px) {
      margin: 0 auto;
    }

    @media (max-width: 500px) {
      width: 60%
    }
  }

  .spotHeading {
    width: 100%;
    color: #2E3C50;
    font-family: Lato;
    font-size: 3rem;
    line-height: 53px;

    @media (max-width: 1000px) {
      font-size: 2rem;
      line-height: 38px;
      text-align: center;
    }

    @media (max-width: 768px) {
      font-size: 2rem;
      line-height: 38px;
      width: 70%;
      margin: 10px 0;
    }

    @media (max-width: 500px) {
      width: 100%;
      margin-top: 15px;
      text-align: center;
      font-size: 1.6rem;
      line-height: 27px;
    }
  }
}

.spotlightCards {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;

  @media (max-width: 1000px) {
    flex-flow: row wrap;
    justify-content: space-evenly;
    width: 70%;
    margin: 0 auto;
  }

  @media (max-width: 768px) {
    width: 80%;
    margin: 0 auto;
  }

  @media (max-width: 650px) {
    width: 100%;
  }

  @media (max-width: 500px) {
    margin: 0 auto;
    flex-flow: column nowrap;
  }
  
  .spotlightCard {
    display: flex;
    flex-direction: column;
    background: #ffffff;
    align-items: center;
    padding: 1%;
    max-width: 225px;
    width: 100%;
    min-height: 300px;
    height: 100%;
    border: none;
    border-radius: 10px;
    box-shadow: -10px 4px 28px rgba(0, 0, 0, 0.25);
    margin: 20px 0;
    transition: all .3s ease;

    &:hover {
      transform: scale(1.05) rotate(-1deg);
    }

    .spotCardImages {
      padding: 5px;
      max-width: 175px;
      height: 150px;
      margin: auto;
      width: 100%;
    }

    .spotDivider {
      margin: 10px auto 0;
      width: 30%;
      border-top: 2px solid #6C63FF;
    }

    .spotCardContent {
      display: flex;
      align-items: center;
      margin: auto;
      width: 100%;
      font-size: 1rem;
      height: 100px;
      font-weight: 400;
      padding: 10px;
    }
  }
}

.btnWrapper {
  display: flex;
  justify-content: flex-end;
  margin: 70px 0 70px 70px;

  @media (max-width: 1000px) {
    justify-content: center;
    margin: 70px auto;
  }

  .spotBtn {
    width: 290px;
    height: 50px;
    font-family: Lato;
    font-size: 1.2rem;
    font-weight: 500;
    box-shadow: -10px 4px 28px rgba(0, 0, 0, 0.25);
    border: 1px solid #6C63FF;
    border-radius: 6px;
    background: #ffffff;
    color: #6C63FF;
    text-align: center;
    cursor: pointer;
    transition: all .3s ease;
    
    &:hover {
      background: #6C63FF;
      color: #fff;
    }
  }
}
