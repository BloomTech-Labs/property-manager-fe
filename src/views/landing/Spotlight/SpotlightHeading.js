import React from 'react';
import SpotHeaderImage from '../../../assets/svg/landing-3homes-middle.svg';
import './spotlight.scss';

const SpotlightHeading = () => {
  return (
    <div className="spotHeadingWrapper">
      <img
        className="spotHeaderImage"
        src={SpotHeaderImage}
        alt="Orange computer and phone"
      />
      <h1 className="spotHeading">
        Manage your properties, donâ€™t let them manage you!
      </h1>
    </div>
  );
};

export default SpotlightHeading;
