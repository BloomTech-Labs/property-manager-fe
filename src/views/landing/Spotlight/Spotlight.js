import React from 'react';
import SpotlightHeading from './SpotlightHeading';
import SpotlightCards from './SpotlightCards';
import SpotlightButton from './SpotlightButton';

const Spotlight = () => {
  return (
    <div className="spotlightWrapper" id="spotlight">
      <SpotlightHeading />
      <SpotlightCards />
      <SpotlightButton />
    </div>
  );
};

export default Spotlight;
