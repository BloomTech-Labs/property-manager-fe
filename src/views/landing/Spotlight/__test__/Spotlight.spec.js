import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom';
import Spotlight from '../Spotlight';

afterEach(rtl.cleanup);

test('Makes sure the word manage exists on SL', async () => {
  const wrapper = rtl.render(<Spotlight />);
  expect(await wrapper.queryAllByText(/manage/i));
});

test('snapshot', () => {
  const { container } = rtl.render(<Spotlight />);
  expect(container.firstChild).toMatchSnapshot();
});
