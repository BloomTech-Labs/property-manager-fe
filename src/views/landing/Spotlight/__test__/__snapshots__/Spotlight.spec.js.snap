// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="spotlightWrapper"
  id="spotlight"
>
  <div
    class="spotHeadingWrapper"
  >
    <img
      alt="Orange computer and phone"
      class="spotHeaderImage"
      src="landing-3homes-middle.svg"
    />
    <h1
      class="spotHeading"
    >
      Manage your properties, don’t let them manage you!
    </h1>
  </div>
  <div
    class="spotlightCards"
  >
    <div
      class="spotlightCard"
    >
      <img
        alt="Spotlight card with documents."
        class="spotCardImages"
        src="landing-docu-card.svg"
      />
      <hr
        class="spotDivider"
      />
      <h5
        class="spotCardContent"
      >
        Communicate and share documents with you tenants directly in the app.
      </h5>
    </div>
    <div
      class="spotlightCard"
    >
      <img
        alt="Spotlight card highlghting Stripe."
        class="spotCardImages"
        src="landing-stripe.svg"
      />
      <hr
        class="spotDivider"
      />
      <h5
        class="spotCardContent"
      >
        Accept payments with confidence through Stripe’s secure API.
      </h5>
    </div>
    <div
      class="spotlightCard"
    >
      <img
        alt="Spotlight card highlighting work orders."
        class="spotCardImages"
        src="landing-workorder.svg"
      />
      <hr
        class="spotDivider"
      />
      <h5
        class="spotCardContent"
      >
        Organize and review repair requests from creation to completion.
      </h5>
    </div>
    <div
      class="spotlightCard"
    >
      <img
        alt="Spotlight card highlighting statistics and charts."
        class="spotCardImages"
        src="landing-chart.svg"
      />
      <hr
        class="spotDivider"
      />
      <h5
        class="spotCardContent"
      >
        Keep track of your cash flow month-to-month so you know exactly where your finances are.
      </h5>
    </div>
  </div>
  <div
    class="btnWrapper"
  >
    <button
      class="spotBtn"
      type="submit"
    >
      Sign up
    </button>
  </div>
</div>
`;
